exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/TDigestFunctions.java

```diff
@@ -0,0 +1,1126 @@
+    @Workspace ObjectHolder work;
+    @Output NullableVarBinaryHolder out;
+    @Inject DrillBuf buffer;
+    @Inject OptionManager options;
+    @Workspace IntHolder compression;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void add() {
+
+    }
+
+    @Override
+    public void output() {
+    }
+
+    @Override
+    public void reset() {
+
+    }
+  }
+
+  @FunctionTemplate(name = "tdigest_merge", scope = FunctionTemplate.FunctionScope.POINT_AGGREGATE)
+  public static class TDigestMergeFunction implements DrillAggFunc {
+    @Param NullableVarBinaryHolder in;
+    @Workspace ObjectHolder work;
+    @Output NullableVarBinaryHolder out;
+    @Inject DrillBuf buffer;
+    @Inject OptionManager options;
+    @Workspace IntHolder compression;
+
+    @Override
+    public void setup() {
+      work = new ObjectHolder();
+      compression.value = (int) options.getLong(org.apache.drill.exec.ExecConstants.TDIGEST_COMPRESSION);
+      work.obj = new com.clearspring.analytics.stream.quantile.TDigest(compression.value);
+    }
+
+    @Override
+    public void add() {
+      if (work.obj != null) {
+        com.clearspring.analytics.stream.quantile.TDigest tdigest = (com.clearspring.analytics.stream.quantile.TDigest) work.obj;
+        try {
+          if (in.isSet != 0) {
+            byte[] buf = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(in.start, in.end, in.buffer).getBytes();
+            com.clearspring.analytics.stream.quantile.TDigest other =
+              com.clearspring.analytics.stream.quantile.TDigest.fromBytes(java.nio.ByteBuffer.wrap(buf));
+            tdigest.add(other);
+          }
+        } catch (Exception e) {
+          throw new org.apache.drill.common.exceptions.DrillRuntimeException("Failed to merge TDigest output", e);
+        }
+      }
+    }
+
+    @Override
+    public void output() {
+      if (work.obj != null) {
+        com.clearspring.analytics.stream.quantile.TDigest tdigest = (com.clearspring.analytics.stream.quantile.TDigest) work.obj;
+        try {
+          int size = tdigest.smallByteSize();
+          java.nio.ByteBuffer byteBuf = java.nio.ByteBuffer.allocate(size);
+          tdigest.asSmallBytes(byteBuf);
+          out.buffer = buffer.reallocIfNeeded(size);
+          out.start = 0;
+          out.end = size;
+          out.buffer.setBytes(0, byteBuf.array());
+          out.isSet = 1;
+        } catch (Exception e) {
+          throw new org.apache.drill.common.exceptions.DrillRuntimeException("Failed to get TDigest output", e);
+        }
+      } else {
+        out.isSet = 0;
+      }
+    }
+
+    @Override
+    public void reset() {
+      work.obj = new com.clearspring.analytics.stream.quantile.TDigest(compression.value);
+    }
+  }
+
+  @FunctionTemplate(name = "tdigest_decode", scope = FunctionScope.SIMPLE, nulls = NullHandling.NULL_IF_NULL)
+  public static class TDigestDecode implements DrillSimpleFunc {
+
```

Comment:
Turns out I am not using this function since the 'decode' of the t-digest happens outside the scope of the generated code.  I will remove this function altogether.